x-pack/legacy/plugins/siem/public/components/ml_popover/jobs_table/filters/toggle_selected_group.tsx

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Dispatch, SetStateAction } from 'react';
+
+export const toggleSelectedGroup = (
+  group: string,
+  selectedGroups: string[],
+  setSelectedGroups: Dispatch<SetStateAction<string[]>>
+) => {
```

Comment:
You've done such a great job of really small concise typescript functions with almost all explicit return types to keep everything very well done!

Don't forget the void on this one:

```suggestion
): void => {
```

This is going to make refactoring and re-working the code so much better seeing these good practices and consistency.

Reply:
Adding -- thanks! 🙌 

As you mentioned, components/business logic becomes significantly easier to rework or refactor when defining strong types and adequately abstracting. It was a pleasure to revisit all this ML code and optimize it a bit more. It's so much easier to identify the data flow now and also way easier to spot and fix bugs as well.